src/p2/element.rs

```diff
@@ -1691,7 +1696,16 @@ pub fn input_from_properties(
         )?,
         placeholder: ftd::p2::utils::string_optional("placeholder", properties, doc.name, 0)?,
         multiline: ftd::p2::utils::bool("multiline", properties, doc.name, 0)?,
-        value: ftd::p2::utils::string_optional("value", properties, doc.name, 0)?,
+        value: {
+            // if value is unchanged then
+            // value = default-value
+            // otherwise retain the changed value
+            match value {
+                Some(v) => Some(v),
+                None => default_value.clone(),
+            }
+        },
```

Comment:
This means even if author doesn't want `value` you are still putting `value`. Remember DOM behaviour differs based on if you have provided `value` or `default-value`. We are implementing `default-value` because we do not want to set `value`. 

Why are you doing this?

Reply:
So you mean if we give default value we need to handle it seperately by generating 
<input type="text", value=default-value, ...> instead of relying on value for generating the same ?

Reply:
If there is `default-value`, the HTML will contain `<input defaultValue="default-value">` (or the JS equivalent), not `value=""`.